### PR TITLE
docs: add explainer for gotchas with NSView based native components

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

While working on https://github.com/microsoft/fluentui-react-native/pull/3026, I found a couple of "gotchas" while writing a native component for `NSVisualEffectView`. This change attempts to documents those gotchas so that others can benefit. Special thanks to @lenaic who  figured out the hit testing issue.

## Changelog

[GENERAL] [ADDED] - Add documentation for gotchas with NSView based native components

## Test Plan

N/A

### CODE REVIEW

**Comment by Saadnajmi** [comment]
Marking as draft to see if I can update RCTUIView hitTest instead

File: docs/WritingNativeComponents.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+			}
+			let result = subview.hitTest(pointForHitTest)
+			if (result != nil) {
+				return result
+			}
+		}
+		return nil
+	}
+}
+``````
```
Comment (tido64) [inline docs/WritingNativeComponents.md:36]: You only need three backticks (and please add a new line)

File: docs/WritingNativeComponents.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+internal class FixedVisualEffectView: NSVisualEffectView {
+
+	/// React Native macOS uses a flipped coordinate space by default. to match the other platforms.
+	/// Let's stay consistent and ensure any views hosting React Native views are also flipped.
+	/// This helps RCTTouchHandler register clicks in the right location, and ensures `layer.geometryFlipped` is true.
+	override var isFlipped: Bool {
+		return true
+	}
+
+	/// This subclass is necessary due to differences' in hitTest()'s implementation between iOS and macOS
```
Comment (tido64) [inline docs/WritingNativeComponents.md:16]:
```suggestion
	/// This subclass is necessary due to differences in hitTest()'s implementation between iOS and macOS
```

File: docs/WritingNativeComponents.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Writing native components derived from NSView
+
+
+React Native macOS inherits some assumptions from React Native on iOS / UIKit. There are subtle differences in how AppKit's and UIKit;s implementation, such as where the coordinate system places the origin (bottom left on Appkit, top left on UIKit), or how hit testing is implemented This serves as an issue when we want to write our own native components derived from NSView, as we don't inherit the "fixes" made in RCTView to get views working properly. This doc illustrates what methods / implementation you will need to override in order for your native component, using the NSView derived `NSVisualEffectView` as our base class
```
Comment (tido64) [inline docs/WritingNativeComponents.md:4]: Punctuation:

```suggestion
React Native macOS inherits some assumptions from React Native on iOS / UIKit. There are subtle differences in how AppKit's and UIKit's implementation, such as where the coordinate system places the origin (bottom left on Appkit, top left on UIKit), or how hit testing is implemented. This serves as an issue when we want to write our own native components derived from NSView, as we don't inherit the "fixes" made in RCTView to get views working properly. This doc illustrates what methods / implementation you will need to override in order for your native component, using the NSView derived `NSVisualEffectView` as our base class.
```

**Pull request merged**